translator/internaldata/metrics_to_oc.go

```diff
@@ -81,12 +82,12 @@ func resourceMetricsToOC(rm pdata.ResourceMetrics) consumerdata.MetricsData {
 	return ocMetricsData
 }
 
-func metricToOC(metric pdata.Metric) *ocmetrics.Metric {
+func metricToOC(metric pdata.Metric, res *ocresource.Resource) *ocmetrics.Metric {
 	labelKeys := collectLabelKeys(metric)
 	return &ocmetrics.Metric{
 		MetricDescriptor: descriptorToOC(metric, labelKeys),
 		Timeseries:       dataPointsToTimeseries(metric, labelKeys),
-		Resource:         nil,
+		Resource:         res,
```

Comment:
I suppose this may lead to less efficient reverse translation in `internaldata.OCToMetrics` (i.e. when we translate back and forth) since metrics with nil resources are combined together as I discovered in #1803.

That seems like a separate issue to me though.